src/region_growing.jl

```diff
@@ -206,3 +206,136 @@ function seeded_region_growing{CT<:Colorant, N}(img::AbstractArray{CT,N}, seeds:
+                       whose ith element is an odd positive integer representing
+                       the length of the ith edge of the N-orthotopic neighbourhood
+* `neighbourhood`   :  (Optional) Function taking CartesianIndex{N} as input and
+                       returning the neighbourhood of that point.
+* `diff_fn`         :  (Optional) Function that returns a difference measure (δ)
+                       between the mean color of a region and color of a point
+
+# Examples
+
+```jldoctest
+julia> img = zeros(Gray{N0f8},4,4);
+julia> img[2:4,2:4] = 1;
+julia> seg = unseeded_region_growing(img, 0.2);
+julia> seg.image_indexmap
+4×4 Array{Int64,2}:
+ 1  1  1  1
+ 1  2  2  2
+ 1  2  2  2
+ 1  2  2  2
+
+```
+
+"""
+function unseeded_region_growing{CT<:Colorant, N}(img::AbstractArray{CT,N}, threshold::Real,
+    kernel_dim::Union{Vector{Int}, NTuple{N, Int}} = [3 for i in 1:N], diff_fn::Function = default_diff_fn)
+    length(kernel_dim) == N || error("Dimension count of image and kernel_dim do not match")
+    for dim in kernel_dim
+        dim > 0 || error("Dimensions of the kernel must be positive")
+        isodd(dim) || error("Dimensions of the kernel must be odd")
+    end
+    pt = CartesianIndex([dim ÷ 2 for dim in kernel_dim]...)
+    unseeded_region_growing(img, threshold, ((c)->CartesianRange(c-pt,c+pt)), diff_fn)
+end
+
+function unseeded_region_growing{CT<:Colorant,N}(img::AbstractArray{CT,N}, threshold::Real, neighbourhood::Function, diff_fn = default_diff_fn)
+   
+    # Required data structures 
+    result                  =   similar(dims->fill(-1,dims), indices(img))      # Array to store labels
+    neighbours              =   PriorityQueue(CartesianIndex{N}, Float64)       # Priority Queue containing boundary pixels with δ as the priority
+    region_means            =   Dict{Int, Images.accum(CT)}()                   # A map containing (label, mean) pairs
+    region_pix_count        =   Dict{Int, Int}()                                # A map containing (label, pixel_count) pairs
+    labels                  =   Vector{Int}()                                   # Vector containing assigned labels
+
+    # Initialize data structures
+    start_point = first(CartesianRange(indices(img)))
+    result[start_point] = 1
+    push!(labels, 1)
+    region_means[1] = img[start_point]
+    region_pix_count[1] = 1
+
+    # Enqueue neighouring points of `start_point`
+    for p in neighbourhood(start_point)
+        if p != start_point && checkbounds(Bool, img, p) && result[p] == -1
+            enqueue!(neighbours, p, diff_fn(region_means[result[start_point]], img[p]))
+        end
+    end
+    
+    while !isempty(neighbours)
+        point = dequeue!(neighbours)
+        δ = Inf
+        minlabel = -1
+        for p in neighbourhood(point)
+            if p != point && checkbounds(Bool, img, p) && result[p] > 0
+                curδ = diff_fn(region_means[result[p]], img[point])
```

Comment:
Trivial, but you could hoist this out, `pixelval = img[point]` above the `for p in neighbourhood(point)` loop.